src/mtcp/mtcp_restart.c

```diff
@@ -524,23 +524,20 @@ static void mtcp_simulateread(int fd, MtcpHeader *mtcpHdr)
     if (area.size == -1) break;
     if ((area.properties & DMTCP_ZERO_PAGE) == 0 &&
         (area.properties & DMTCP_SKIP_WRITING_TEXT_SEGMENTS) == 0) {
-      void *addr = mtcp_sys_mmap(0, area.size, PROT_WRITE | PROT_READ,
-                                 MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
-      if (addr == MAP_FAILED) {
-        MTCP_PRINTF("***Error: mmap failed; errno: %d\n", mtcp_sys_errno);
-        mtcp_abort();
+
+      if (area.flags & MAP_SHARED) {
+        area.size =  area.mmapFileSize;
```

Comment:
Suppose `area.mmapFileSize == 0` (in your version of the code) or e `area.mmapFileSize == -1` (with my proposed change).  This is the case where we issued:..
`JWARNING(false)(JASSERT_ERRNO)(area->name).Text("Error getting file info");`

We should probably just use the area.size in the ckpt image file and not try to overwrite it.